pkg/monitoring/alloy/secret.go

```diff
@@ -16,9 +27,51 @@ const (
 
 var (
 	//go:embed templates/monitoring-secret.yaml.template
-	alloyMonitoringSecret string
+	alloyMonitoringSecret         string
+	alloyMonitoringSecretTemplate *template.Template
 )
 
 func init() {
-	template.Must(template.New("monitoring-secret.yaml").Funcs(sprig.FuncMap()).Parse(alloyMonitoringSecret))
+	alloyMonitoringSecretTemplate = template.Must(template.New("monitoring-secret.yaml").Funcs(sprig.FuncMap()).Parse(alloyMonitoringSecret))
+}
+
+func (a *Service) GenerateAlloyMonitoringSecretData(ctx context.Context, cluster *clusterv1.Cluster) (map[string][]byte, error) {
```

Comment:
Do we not need an interface on top of this to define whether to enable alloy config or prometheus agent based on the olly bundle version?

Reply:
The logic to decide between alloy or prometheus agent will be outside of this package.

Reply:
ack